### PR TITLE
Stoichiometry + chemistry fixes, full docs overhaul, 99% test coverage

### PR DESCRIPTION
**Pull request by timlichtenberg**
**Update 2026-05-03 (post-adversarial review): three further hardening commits on top of `e7efa2e`. All defaults preserved; PROTEUS callers see no change. 126/126 tests pass (+35 new since the docs review), 99% coverage on changed files.**
- **`1e99d48`** — Harden `p_guess` validator: reject non-dict types (list, tuple, Series) with a clear `TypeError`; reject non-finite values (NaN, ±Inf) with a `ValueError` naming the offending key. Previously these slipped past the missing-key check and either raised an opaque `TypeError` from the membership test, or fed `NaN` into `fsolve` and produced silent garbage output. 10 new parametrized tests cover both paths.
- **`36cfabe`** — Skip `dasfac_2 = exp(...)` precompute in `SolubilityN2.__init__` when `composition='libourel'`. Previously the precompute ran unconditionally, so extreme `x_SiO2` values overflowed `np.exp` and emitted `RuntimeWarning` at construction time even on the libourel path that never reads the value. Updated the libourel-with-extreme-input test to assert *no* warning is emitted (via `warnings.simplefilter('error')`).
- **`ca87ecc`** — Test polish: drop a redundant `out > 0.0` assertion that was trivially implied by the preceding `math.isfinite` (creating false confidence in sign-bug detection); add a `Hidden coupling` note at both S2 regression pin sites so a future audit of `oxygen_fugacity.py` IW coefficients sees that `test_solubility_compositions.py` is also load-bearing.

**Earlier update 2026-05-03 (post @nichollsh review): four additions on top of the merged docs suggestions in `d88f37b`. All non-default behaviour is opt-in; PROTEUS callers see bit-identical numerics.**
- **`a2be794`** — Cite Clark (1982), *Carbon Dioxide Review*, p. 469, Oxford Univ. Press as the source of `ocean_moles` in `src/calliope/constants.py` (comment-only; numerical value unchanged).
- **`ab2c26d`** — Validate `p_guess` keys in `equilibrium_atmosphere`. Missing any of `'H2O'`, `'CO2'`, `'N2'`, `'S2'` now raises a clear `ValueError` naming the missing keys, instead of a bare `KeyError` from inside the tuple construction. Six new unit tests in test_equilibrium_paths.py: parametrized missing-key, empty-dict, extra-keys-allowed.
- **`20db876`** — Expose `x_FeO` as a `SolubilityS2` constructor kwarg (default `10.0` wt%, the prior hardcoded Earth-mantle value). Non-Earth bulk compositions can now configure it via `SolubilityS2(x_FeO=...)`. Five new unit tests pin the default behaviour and exercise the closed-form `exp(0.124 · Δx_FeO)` ratio against plausible coefficient/sign bugs.
- **`e7efa2e`** — Expose `x_SiO2`, `x_Al2O3`, `x_TiO2` as `SolubilityN2` constructor kwargs (defaults `0.56, 0.11, 0.01`, the prior hardcoded Earth-mantle values). `dasfac_2` stays precomputed in `__init__`. Ten new unit tests pin the default `dasfac_2 ≈ 406.79` and parametrize per-kwarg scaling against each of the three published coefficients (`7.11`, `−13.06`, `−120.67`) independently. `docs/Explanations/solubility.md` updated; the prior `c_melt ≈ 408` estimate corrected to `≈ 407` to match the actual computed value.

---

## Description

This branch started as a fix for issue #16 (incorrect elemental stoichiometry, equilibrium constants, and CH4 solubility) and grew into a wider cleanup of CALLIOPE. This is the extent of functionality changes I want to do with this PR, anything else would be a different PR, with different documentation and test additions. It is part of the larger `tl/interior-refactor` PROTEUS-wide refactor. The other modules will follow in the upcoming days. 

closes #16

Three themes:

1. **Physics correctness fixes** (the original issue):
   - Elemental stoichiometry, equilibrium-constant signs, and CH4 solubility coefficient corrected.
   - `OxygenFugacity` now raises `ValueError` on `T <= 0` instead of returning NaN (both IW formulae diverge there).
   - Redundant equilibrium-chemistry call inside `equilibrium_atmosphere` removed (`sol_dict` is now reused for `atmosphere_mass` / `dissolved_mass`).
   - `corefrac` parameter renamed to `core_frac` for readability.
   - Unreachable `Invalid solver` branch removed from `equilibrium_atmosphere`.

2. **Documentation overhaul** (Diataxis layout, MkDocs Material via Zensical):
   - New top-level pages: `getting_started.md`, `proteus-framework.md`, restructured `index.md`.
   - Tutorials: first-run walk-through with an Earth-like solve.
   - How-to: installation, configuration, usage, PROTEUS coupling, testing, releasing.
   - Explanations: model overview, equilibrium chemistry, solubility laws, oxygen fugacity, mass balance, code architecture, PROTEUS coupling theory.
   - Reference: mkdocstrings API stubs for all six modules; curated publications page.
   - Every paper mention is now an inline ADS or DOI link, audited end-to-end.

3. **Test coverage and CI**:
   - Coverage 81% → 99%, hard floor pinned at 95% in `pyproject.toml`.
   - Four new test files: `test_partial_species.py`, `test_targets.py`, `test_equilibrium_paths.py`, expanded `test_stoichiometry.py`.
   - Two deliberate-edge-case tests now use `pytest.warns(RuntimeWarning)` so the suite exits cleanly.
   - Codecov wired up via the org-wide token; legacy gist-driven badge removed from README and docs.
   - Workflow renamed `CALLIOPE CI Test Suite`.

Other housekeeping:

- CalVer version bump 25.05.04 → 26.05.02 across `pyproject.toml`, `CITATION.cff`, `__init__.py`.
- README rewritten from the ground up; Python-versions badge dropped.
- Docs deploy migrated to Zensical + GitHub Pages actions.
- This PR template added (mirrors the PROTEUS one).

## Validation of changes

- Test configuration: macOS 15.3 (arm64) with Python 3.12 (conda env `proteus`), pytest 8.4.1, ruff 0.12.4.
- `pytest tests/`: 88 passed, 6 warnings (all from the deliberate `test_hide_warnings_false_lets_warnings_through` path or third-party scipy `trust-constr` noise).
- `coverage report --format=total`: 99% (above the 95% floor).
- `ruff check src tests` and `ruff format --check src tests`: clean.
- `zensical build --clean`: builds in 1.13 s with no errors.
- Branch parity vs `main`: 43 commits ahead, +4401 / -572 across 56 files.

## Checklist

- [x] I have followed the [contributing guidelines](https://proteus-framework.org/CALLIOPE/Community/CONTRIBUTING.html)
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings or errors
- [x] I have checked that the tests still pass on my computer
- [x] I have updated the docs, as appropriate
- [x] I have added tests for these changes, as appropriate
- [x] I have checked that all dependencies have been updated, as required

## Relevant people

@nichollsh

### CODE REVIEW

File: src/calliope/structure.py [hunk 1]
```diff
@@ -7,29 +7,31 @@
 
 from .constants import M_earth, R_earth
 
-log = logging.getLogger("fwl."+__name__)
+log = logging.getLogger('fwl.' + __name__)
 
 
-def calculate_mantle_mass(radius:float, mass:float, corefrac:float)->float:
-    '''
+def calculate_mantle_mass(radius: float, mass: float, core_frac: float) -> float:
```
[the remark this answers is not in the document]
Comment (timlichtenberg) [reply src/calliope/structure.py]: Resolved in fe4612a: `calculate_mantle_mass()` now accepts `corefrac` as a keyword-only alias that emits `DeprecationWarning` and forwards to `core_frac`. Three regression tests cover the alias warning, the both-aliases conflict, and the missing-argument error.

File: docs/Explanations/model.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+## Validity range
+
+CALLIOPE is calibrated for surface temperatures of roughly $1000 \le T_\mathrm{magma} \le 4000$ K and surface pressures of roughly $0.1 \le p_\mathrm{surf} \le 5000$ bar. The lower end of the pressure range is set by numerical stability of the speciation walk; the upper end is the loose envelope above which one or more solubility laws extrapolate. Individual solubility laws have tighter calibration windows than the envelope (Dixon CO$_2$: $\le$815 bar; Sossi H$_2$O: a few kbar; Ardia CH$_4$: 0.7-3 GPa total pressure), see the per-law table in [Solubility laws](solubility.md). Outside the envelope above:
+
+- Below $T \sim 1000$ K the JANAF fits used for the equilibrium constants extrapolate beyond their validation range; the PROTEUS wrapper enforces a configurable `T_floor` (default 700 K) below which outgassing is skipped.
```
Comment (nichollsh) [inline docs/Explanations/model.md]:
```suggestion
- Below $T \sim 1000$ K the JANAF fits used for the equilibrium constants extrapolate beyond their validation range. The PROTEUS wrapper enforces a configurable `T_floor` (default 700 K), which clips temperatures below `T_floor` to this value, since thermochemical equilibrium does not necessarily hold at cooler temperatures.
```

File: docs/Explanations/model.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+- Above $T \sim 4000$ K the gas-phase ideal-gas approximation breaks down; switch to atmodeller ([Bower et al. 2025](https://ui.adsabs.harvard.edu/abs/2025ApJ...995...59B)), which uses real-gas EOS.
```
Comment (nichollsh) [inline docs/Explanations/model.md]: I agree that there are complexities at high T, but this is not the main one. The ideal gas EOS is more valid at high temperatures and low pressures. 

```suggestion
- Above $T \sim 4000$ K the mantle-atmosphere partitioning approximation breaks down; switch to atmodeller ([Bower et al. 2025](https://ui.adsabs.harvard.edu/abs/2025ApJ...995...59B)).
```

File: docs/Explanations/oxygen_fugacity.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+## Reference values for $\Delta\mathrm{IW}$
+
+| Reservoir | $\Delta\mathrm{IW}$ | Source |
+|---|---|---|
+| Mercury surface | $\sim -5$ | [Cartier & Wood (2019)](https://ui.adsabs.harvard.edu/abs/2019Eleme..15...39C) |
+| Asteroidal material | $\sim -2$ | [Doyle et al. (2019)](https://ui.adsabs.harvard.edu/abs/2019Sci...366..356D) |
+| Mars mantle | $-3$ to $0$ | [Wadhwa (2001)](https://ui.adsabs.harvard.edu/abs/2001Sci...291.1527W) |
+| Iron-wüstite buffer | $0$ | by definition |
+| Earth's mantle $f_{\mathrm{O}_2}$ | $+3.5 \pm 0.5$ | [Sossi et al. (2020)](https://ui.adsabs.harvard.edu/abs/2020SciA....6.1387S) |
```
Comment (nichollsh) [inline docs/Explanations/oxygen_fugacity.md]: Correction below reflects what the paper actually states.
```suggestion
| Earth's upper mantle $f_{\mathrm{O}_2}$ | $+3.5$ | [Sossi et al. (2020)](https://ui.adsabs.harvard.edu/abs/2020SciA....6.1387S) |
```

File: src/calliope/solubility.py [hunk 5]
```diff
@@ -67,16 +73,16 @@ def gaillard(self, p, temp, fO2_shift):
             return 0.0
 
         # melt composition [wt%]
-        x_FeO  = 10.0
```
Comment (nichollsh) [inline src/calliope/solubility.py:70]: Perhaps this is a good time to expose this hidden (but important) composition variable as a parameter?

File: src/calliope/solubility.py [hunk 6]
```diff
@@ -101,42 +107,43 @@ def __init__(self, composition='libourel'):
         super().__init__(composition)
 
         # melt composition
-        x_SiO2  = 0.56
+        x_SiO2 = 0.56
         x_Al2O3 = 0.11
-        x_TiO2  = 0.01
-        self.dasfac_2 = np.exp(4.67 + 7.11*x_SiO2 - 13.06*x_Al2O3 - 120.67*x_TiO2)
+        x_TiO2 = 0.01
```
Comment (nichollsh) [inline src/calliope/solubility.py]: Similarly, to FeO, maybe these could be exposed as parameters too.

File: src/calliope/solve.py [hunk 7]
```diff
@@ -276,328 +259,319 @@ def dissolved_mass(pin, ddict):
...
+    # non-physical regions during a poor cold start.
+    lb = [0.0] * 4
+    ub = [1e7] * 4
 
-    # Initial guess for partial pressure [bar]
     if p_guess is None:
         x0 = get_initial_pressures(target_d)
     else:
-        x0 = (p_guess["H2O"], p_guess["CO2"], p_guess["N2"], p_guess["S2"])
+        x0 = (p_guess['H2O'], p_guess['CO2'], p_guess['N2'], p_guess['S2'])
```
Comment (nichollsh) [inline src/calliope/solve.py:444]: This should probably check that `p_guess` has the four molecules' names as dictionary keys. I think it will throw a KeyError if `p_guess!=None` but (e.g.) `p_guess` is missing `"H2O"`.

File: src/calliope/constants.py [hunk 8]
```diff
@@ -4,53 +4,53 @@
...
-M_earth         = 5.972E24              # kg
-R_earth         = 6.335439e6
-R_core_earth    = 3485000.0             # m
-M_core_earth    = 1.94E24               # kg
-ocean_moles     = 7.68894973907177e+22 # moles of H2 (or H2O) in one present-day Earth ocean
+M_earth = 5.972e24  # kg
+R_earth = 6.335439e6
+R_core_earth = 3485000.0  # m
+M_core_earth = 1.94e24  # kg
+ocean_moles = 7.68894973907177e22  # moles of H2 (or H2O) in one present-day Earth ocean
```
Comment (nichollsh) [inline src/calliope/constants.py]: The reference for this value is below. Maybe worth adding into the docs/code.

```
@book{clark_carbon_1982,
    author = {{Clark}, W. C.},
    booktitle = "Carbon Dioxide Review",
    year = 1982,
    pages = 469,
    address = {New York},
    publisher = {Oxford University Press},
}
```

**Review by nichollsh — CHANGES_REQUESTED** [review]
Very nice! Thank you for these fixes and improvements, @timlichtenberg.

Fantastic that the test coverage is now so high. They pass on my laptop successfully.

I have also verified that these CALLIOPE updates work with PROTEUS simulations. Importantly, I also checked how much the updated chemistry+solubility coefficients change the simulations. The biggest difference is that SO2 and S2 are less abundant in the atmosphere, but seems like H2S is hardly impacted, so I am not concerned about any critical implications for publications using old versions of CALLIOPE.

The docs build fine on my laptop and look great. Nice to have CALLIOPE so well documented now. 

* A few comments about the wording used in some of the docs.
* Added some citation suggestions. 
* I suggest exposing some important (but hidden) variables as parameters.

File: src/calliope/constants.py (hunk 8 above)
Comment (timlichtenberg) [reply src/calliope/constants.py]: Done in a2be794: added the Clark, W. C. (1982), Carbon Dioxide Review, p. 469, Oxford Univ. Press source as a comment alongside ocean_moles in src/calliope/constants.py. Numerical value unchanged.

File: src/calliope/solve.py (hunk 7 above)
Comment (timlichtenberg) [reply src/calliope/solve.py:444]: Done in ab2c26d: equilibrium_atmosphere now validates that p_guess contains all four primary-species keys ('H2O', 'CO2', 'N2', 'S2') up front and raises ValueError naming the missing keys, instead of falling through to a bare KeyError from the tuple construction. Six new unit tests in test_equilibrium_paths.py: parametrized missing-key cases for each primary, the empty-dict edge case, and a positive test that extra keys (e.g. diagnostic species like SO2) are accepted.

File: src/calliope/solubility.py (hunk 5 above)
Comment (timlichtenberg) [reply src/calliope/solubility.py:70]: Done in 20db876: x_FeO is now a SolubilityS2 constructor kwarg (default 10.0 wt%, the prior hardcoded Earth-mantle value). PROTEUS callers using SolubilityS2() with no arguments see bit-identical numerics; non-Earth bulk compositions can now configure it via SolubilityS2(x_FeO=...). Five new unit tests in test_solubility_compositions.py pin the default behaviour, exercise the closed-form exp(0.124 * dx_FeO) ratio across x_FeO in {5, 10, 15, 20} wt%, and cover the zero-pressure short-circuit + extreme-value edges. docs/Explanations/solubility.md updated to document the kwarg.

File: src/calliope/solubility.py (hunk 6 above)
Comment (timlichtenberg) [reply src/calliope/solubility.py]: Done in e7efa2e: x_SiO2, x_Al2O3, x_TiO2 are now SolubilityN2 constructor kwargs (defaults 0.56, 0.11, 0.01 — the prior hardcoded Earth-mantle values). dasfac_2 stays precomputed in __init__ to preserve the fast-path for Monte-Carlo restart loops. PROTEUS callers using SolubilityN2('dasgupta') see bit-identical numerics. Ten new unit tests in test_solubility_compositions.py pin the default dasfac_2 (~ 406.79), parametrize per-kwarg scaling against each of the three coefficients (7.11, -13.06, -120.67) independently, and verify the libourel path is unaffected even with extreme composition values. docs/Explanations/solubility.md updated. The doc text now reads c_melt ~ 407 (the prior ~408 estimate was slightly high).

**Review by nichollsh — APPROVED** [review]
Looks good! I can see you've addressed my previous comments and added tests for these. Happy for this to be merged.

I assume that PROTEUS tests will fail when pulling CALLIOPE, after a CALLIOPE release is made, since this PR will lead to quantitatively different calculations in integration tests.
